lib/puppet/type.rb

```diff
@@ -1289,65 +1289,18 @@ def pathbuilder
   end
 
   newmetaparam(:audit) do
-    desc "(This metaparameter is deprecated and will be ignored in a future release.)
-
-      Marks a subset of this resource's unmanaged attributes for auditing. Accepts an
-      attribute name, an array of attribute names, or `all`.
-
-      Auditing a resource attribute has two effects: First, whenever a catalog
-      is applied with puppet apply or puppet agent, Puppet will check whether
-      that attribute of the resource has been modified, comparing its current
-      value to the previous run; any change will be logged alongside any actions
-      performed by Puppet while applying the catalog.
-
-      Secondly, marking a resource attribute for auditing will include that
-      attribute in inspection reports generated by puppet inspect; see the
-      puppet inspect documentation for more details.
-
-      Managed attributes for a resource can also be audited, but note that
-      changes made by Puppet will be logged as additional modifications. (I.e.
-      if a user manually edits a file whose contents are audited and managed,
-      puppet agent's next two runs will both log an audit notice: the first run
-      will log the user's edit and then revert the file to the desired state,
-      and the second run will log the edit made by Puppet.)"
+    desc "This metaparameter is ignored."
 
     validate do |list|
       if Puppet.settings[:strict] != :off
         # Only warn if `audit` metaparam came from a manifest
         if file && line
-          puppet_deprecation_warning(_("The `audit` metaparameter is deprecated and will be ignored in a future release."), { :line => line, :file => file })
-        end
-      end
-      list = Array(list).collect {|p| p.to_sym}
-      unless list == [:all]
-        list.each do |param|
-          next if @resource.class.validattr?(param)
-          fail "Cannot audit #{param}: not a valid attribute for #{resource}"
+          Puppet.warn(_("The `audit` metaparameter is ignored."), { :line => line, :file => file })
         end
       end
     end
 
     munge do |args|
-      properties_to_audit(args).each do |param|
-        next unless resource.class.validproperty?(param)
-        resource.newattr(param)
-      end
-    end
-
-    def all_properties
```

Comment:
Grepping a forge dump is about the only way I know of to look in to this stuff. It's always a pain.

I think this would be easy enough for somebody to re-implement if they really need it - it's probably not worth going to the effort.